### DISCUSSION TITLE
Class suppressions for Spark doesn't work

### DISCUSSION THREAD

**Discussion by stevensim226** [open-telemetry/opentelemetry-java-instrumentation#2872] [Q&A]
Hello,
I am trying to use OpenTelemetry's auto instrumentation agent to export traces for my microservices modules using Spark java deployed in a Kubernetes cluster which has a healthcheck system by sending calls to `/ping` which unfortunately also bloats my external storage (ElasticSearch) for storing the ping trace data.

I found something in your docs about [trace suppressions](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/suppressing-instrumentation.md) which I have tried using this system properties argument when running the `.jar` file with the java agent included = `-Dotel.javaagent.exclude-classes="com.some.package.path.PingController"` but it doesn't seem to suppress  the call to `/ping` (contained in that class).

Is there anything that I forgot to add?

**Comment by stevensim226** [comment]
I think I posted on the wrong repository discussion as this is for manual instrumentation while my case is for auto instrumentation, sorry for that

**Comment by trask** [comment]
excluding `com.some.package.path.PingController` will only exclude instrumentation on that specific class, which means it won't capture the internal controller span, but it will still capture the request span, which is captured upstream at the servlet/filter.

probably you will need #1060

**Reply by stevensim226** [reply]
Hello @trask, thanks for the feedback
Sorry for asking the same question but, has the feature already been implemented for the issue that you have linked?

Or maybe is there a workaround into suppressing some operation / endpoint from getting traced? (without switching to manual instrumentation of course)

**Reply by trask** [reply]
#1060 hasn't been implemented yet.

You could create a custom distribution of the javaagent (e.g. https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/distro) and add a Sampler which implements this suppression logic. This is a bit complex for something that should be simple though.

Adding a custom Sampler should be easier with #2442 (also not implemented yet, though in progress).
